### PR TITLE
IT-4330 tile component

### PR DESCRIPTION
**Pull request by pioziol123**

### CODE REVIEW

File: lib/js/atoms/Tile/Tile.vue [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+import { VueConstructor } from 'vue';
+
+export default {
+	name: 'Tile',
+	components: {
+		WnlIcon,
+		RippleWrapper,
+	},
+	props: {
+		clickable: {
```
Comment (kvas-damian) [inline lib/js/atoms/Tile/Tile.vue:108]: Props doesn't match what we agreed on in https://miro.com/app/board/uXjVOaIWGa0=/?moveToWidget=3458764516053146935&cot=14

File: lib/js/atoms/Tile/Tile.vue [hunk 2]
```diff
@@ -0,0 +1,140 @@
+<template>
+	<ripple-wrapper :disable="!clickable">
+		<div :class="{ '-clickable': clickable }" class="a-tile">
+			<div v-if="left" class="a-tile__left">
```
Comment (kvas-damian) [inline lib/js/atoms/Tile/Tile.vue:4]: do we need additional `<div>`. Maybe we could apply class on icon?

File: lib/js/atoms/Tile/Tile.vue [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+		@include textM;
+
+		color: $color-neutral-text-heavy;
+	}
+}
+</style>
+
+<script lang="ts">
+import RippleWrapper from '../../utils/RippleWrapper.vue';
+import WnlIcon, { ICON_SIZES, ICONS } from '../Icon';
```
Comment (kvas-damian) [inline lib/js/atoms/Tile/Tile.vue:98]: try to avoid using `wnl` in DS ;)

File: lib/js/atoms/Tile/Tile.vue [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+			</div>
+			<div class="a-tile__center">
+				<div class="a-tile__eyebrow">
+					<slot name="eyebrow" />
+				</div>
+				<div class="a-tile__title">
+					<slot />
+				</div>
+			</div>
+			<div v-if="clickable" class="a-tile__clickableIcon">
```
Comment (kvas-damian) [inline lib/js/atoms/Tile/Tile.vue:15]: unnecessary div?

File: lib/js/atoms/Tile/Tile.vue [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+			type: String,
+			required: false,
+			default: null,
+		},
+		interactive: {
+			type: Boolean,
+			required: false,
+			default: false,
+		},
+		leftIcon: {
```
Comment (kvas-damian) [inline lib/js/atoms/Tile/Tile.vue:129]: in [Miro](https://miro.com/app/board/uXjVOaIWGa0=/?moveToWidget=3458764516053091596&cot=14) and `Button` component we defined this prop as `iconLeft`. What was the reason to change it?

File: lib/js/atoms/Tile/Tile.vue [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+	},
+	props: {
+		text: {
+			type: String,
+			required: true,
+		},
+		eyebrowText: {
+			type: String,
+			required: false,
+			default: null,
```
Comment (kvas-damian) [inline lib/js/atoms/Tile/Tile.vue:122]:
```suggestion
			default: null,
```
it's enough to define one attribute `defualt` or `required`. The same for props below

File: lib/js/atoms/Tile/Tile.vue [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+		margin-right: $space-xs;
+	}
+
+	&__center {
+		display: flex;
+		flex-direction: column;
+		flex-grow: 1;
+	}
+
+	&__interactiveIcon {
```
Comment (kvas-damian) [inline lib/js/atoms/Tile/Tile.vue:69]: IMO we should use `__rightIcon` class with modifier `-interactive` to change the color. By doing that you group common styles in the same place, so there is no way they get out of synch

File: lib/js/atoms/Tile/Tile.vue [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+	</ripple-wrapper>
+</template>
+
+<style lang="scss" scoped>
+@import '../../../styles/settings/typography';
+@import '../../../styles/settings/spacings';
+@import '../../../styles/settings/colors/tokens';
+
+::v-deep .ripple {
+	background-color: $color-neutral-ripple !important;
```
Comment (kvas-damian) [inline lib/js/atoms/Tile/Tile.vue:39]: neutral? Are you sure?

File: lib/js/atoms/Tile/Tile.vue [hunk 9]
```diff
@@ -0,0 +1,156 @@
...
+}
+
+.a-tile {
+	width: 100%;
```
Comment (kvas-damian) [inline lib/js/atoms/Tile/Tile.vue:43]: do we need this `width`? `display:flex` make element full with by default

File: lib/js/atoms/Tile/Tile.vue [hunk 10]
```diff
@@ -0,0 +1,156 @@
...
+	min-height: $space-xl;
```
Comment (kvas-damian) [inline lib/js/atoms/Tile/Tile.vue:44]: `min-height` is suspected. Usually, height of the element is computed from padding and what's inside

File: lib/js/atoms/Tile/Tile.vue [hunk 11]
```diff
@@ -0,0 +1,156 @@
...
+	background-color: $color-neutral-background;
+	display: flex;
+	flex-direction: row;
+	align-items: center;
```
Comment (kvas-damian) [inline lib/js/atoms/Tile/Tile.vue:48]: we try to keep alpabetical order in CSS

File: lib/js/atoms/Tile/Tile.spec.ts [hunk 12]
```diff
@@ -0,0 +1,136 @@
...
+
+describe('Tile', () => {
+	it('Is interactive without left icon', () => {
+		const props = {
+			text: 'text text',
+			eyebrowText: 'eyebrowText text',
+			additionalText: 'additionalText text',
+			rightIcon: ICONS.FA_ANGLE_RIGHT,
+		};
+		const component = createComponent({ ...defaultProps, ...props });
```
Comment (kvas-damian) [inline lib/js/atoms/Tile/Tile.spec.ts:42]: you can set default values in `createComponent()` method

File: lib/js/atoms/Tile/Tile.vue [hunk 13]
```diff
@@ -0,0 +1,139 @@
...
+
+<script lang="ts">
+import RippleWrapper from '../../utils/RippleWrapper.vue';
+import Icon, { ICON_SIZES, ICONS } from '../Icon';
+import { VueConstructor } from 'vue';
+
+export default {
+	name: 'Tile',
+	components: {
+		Icon,
```
Comment (kvas-damian) [inline lib/js/atoms/Tile/Tile.vue:111]: https://v2.vuejs.org/v2/style-guide/?redirect=true#Multi-word-component-names-essential

File: lib/js/atoms/Tile/Tile.spec.ts (hunk 12 above)
Comment (pioziol123) [reply lib/js/atoms/Tile/Tile.spec.ts:42]: Nope. Linter throws errors.

Comment (kvas-damian) [reply lib/js/atoms/Tile/Tile.spec.ts:38]: You need to know how to talk with them :)
https://github.com/bethinkpl/design-system/pull/103/files#diff-6812c905c4c4fd5574700c076c2a2f6d74cf66935ba2ec312bf102fbde421b62R15-R22

File: lib/js/atoms/Tile/Tile.vue [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+
+		&:hover {
+			background-color: $color-neutral-background-hovered;
+		}
+	}
+
+	&__interactiveIcon {
+		color: $color-primary-icon;
+		margin-left: $space-xs;
+	}
```
Comment (kvas-damian) [inline lib/js/atoms/Tile/Tile.vue:90]:
```suggestion
```

**Review by kvas-damian — APPROVED** [review]

**Pull request merged**